clang/runtime/dpct-rt/include/dpl_extras/vector.h

```diff
@@ -258,61 +337,75 @@ class device_vector {
   void shrink_to_fit(void) {
     if (_size != capacity()) {
       size_type tmp_capacity = ::std::max(_size, _min_capacity());
-      auto tmp = _alloc.allocate(tmp_capacity);
+      auto tmp =
+          ::std::allocator_traits<Allocator>::allocate(_alloc, tmp_capacity);
       if (_size > 0) {
         ::std::copy(
             oneapi::dpl::execution::make_device_policy(get_default_queue()),
             begin(), end(), tmp);
       }
-      _alloc.deallocate(_storage, _capacity);
+      ::std::allocator_traits<Allocator>::deallocate(_alloc, _storage,
+                                                     _capacity);
       _storage = tmp;
       _capacity = tmp_capacity;
     }
   }
   void assign(size_type n, const T &x) {
     resize(n);
     if (_size > 0) {
-      ::std::fill(oneapi::dpl::execution::make_device_policy(get_default_queue()),
-                begin(), begin() + n, x);
+      ::std::fill(
```

Comment:
Should we do `clear()` at first?
And why we use `std::fill` instead of `allocator_traits::construct` here?

Reply:
So what will the dpl algorithm do when it tries to replace existed element? Do the inplace construct or assignment?